Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -299,11 +299,30 @@ function Wait-ServiceFabricApplicationUpgradeAction
     }
 
     $upgradeStatusValidator = { param($upgradeStatus) return !($upgradeStatus.UpgradeState -eq "RollingBackCompleted" -or $upgradeStatus.UpgradeState -eq "RollingForwardCompleted") }
+
+    $exceptionRetryEvaluator = {
+        param($ex)
+
+        if ($ex.GetType().FullName -eq "System.Fabric.FabricException")
+        {
+            if($ex.ErrorCode -eq "InvalidCredentials")
```

Comment:
How are we handling the scenario when the failure is genuinely due to bad credentials

Reply:
In that case we'll not reach till this point. Also, I believe that we don't need to handle that case since it will be a valid case of exception and must be thrown/reported.

Reply:
Is it because the task would have failed much earlier?

Reply:
Yes, task would not even able to connect to the cluster on the first place